src/components/UsernameInput/UsernameInput.tsx

```diff
@@ -13,14 +12,35 @@ interface UsernameInputProps extends Omit<TextStyleProps, "onColorChange"> {
 }
 
 const UsernameInput: FC<UsernameInputProps> = ({ state, onInputChange, label }) => {
+    const [emailValidation, setEmailValidation] = useState<"valid" | "invalid">();
+    const emailRegex = new RegExp(/^[A-Za-z0-9._%+-]+@[A-Za-z0-9.-]+\.[A-Za-z]{2,}$/);
+
+    const validateEmail = (input: string) => {
+        setEmailValidation(emailRegex.test(input) ? "valid" : "invalid");
+        //console.log(emailValidation); // testing but output is behind one cycle.
+    };
+
     const handleButtonChange = (options: { isBold?: boolean; isItalics?: boolean }) => {
         // Since the provider needs a type to know where to apply the logic, we are adding it to the options object
         const modifiedOptions = { ...options, type: state.variant };
         onInputChange(modifiedOptions);
     };
 
     const handleInputChange = (value: string) => {
-        onInputChange({ variant: state.variant, textValue: value });
+        if (state.variant === "email") {
+            if (value === "") {
+                setEmailValidation(undefined);
+            } else {
+                validateEmail(value);
+                const isValid: string | undefined = emailValidation; //setting value in check constant.
+                if (isValid === "valid") {
+                    onInputChange({ variant: state.variant, textValue: value });
+                } else {
+                    onInputChange({ variant: state.variant, textValue: undefined });
+                }
+            }
+        }
+        // onInputChange({ variant: state.variant, textValue: value });
```

Comment:
Este si lo vamos a querer acá para que el onInputChange corra as it was before para todos los otros input variants 
```suggestion
        } else {
          onInputChange({ variant: state.variant, textValue: value });
        }
```

Reply:
oh snap se me fue! uncommented!